typescript/src/codeActions/custom/renameToTypeParameter.ts

```diff
@@ -0,0 +1,36 @@
+import { CodeAction } from '../getCodeActions'
+import { getChangesTracker } from '../../utils'
+
+export default {
+    id: 'renameToTypeParameter',
+    name: 'Rename to Type Parameter',
+    kind: 'refactor.rewrite.renameToTypeParameter',
+    tryToApply(sourceFile, position, range, node, formatOptions, languageService) {
+        if (!node || !position) return
+        if (!ts.isIdentifier(node) || !ts.isParameter(node.parent) || !ts.isFunctionLike(node.parent.parent)) return
+
+        const functionParameters = node.parent.parent.parameters
+        const parameterIndex = functionParameters.indexOf(node.parent)
+
+        const functionDecl = node.parent.parent.parent
+        const typeChecker = languageService.getProgram()!.getTypeChecker()
+        const typeAtLocation = typeChecker.getTypeAtLocation(functionDecl)
+
+        const typeDecl = typeAtLocation.getSymbol()?.declarations?.[0]
```

Comment:
I don't think it gets type in all cases, anyway we have already good working logic for getting signature in method snippets: https://github.com/zardoy/typescript-vscode-plugins/blob/3009d9f0085981285fcc4269bbfd4257a6792397/typescript/src/constructMethodSnippet.ts#L21-L24 (you can also extract that code with extract to function refactoring and then export) below there you can see example with getting correct signature, we can just grad the first one for now I think.

Reply:
@zardoy Most probably I didn't get you right. Can you check the code in the last commit?

Reply:
I mean why not use `checker.getSignaturesOfType(type)`? Or you think the current solution is good enough? Will it work with something like this: `const a: ((a, c) => any) & ((b) => any) => {}`?

Reply:
well, I just didn't know  `checker.getSignaturesOfType` exists :)